rest/config.go

```diff
@@ -1082,19 +1067,31 @@ func (sc *ServerContext) fetchAndLoadConfigs() (count int, err error) {
 	return sc._applyConfigs(fetchedConfigs), nil
 }
 
-// fetchAndLoadDatabase will attempt to find the given database name first in a matching bucket name,
-// but then fall back to searching through configs in each bucket to try and find a config.
 func (sc *ServerContext) fetchAndLoadDatabase(dbName string) (found bool, err error) {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+	return sc._fetchAndLoadDatabase(dbName)
+}
+
+// _fetchAndLoadDatabase will attempt to find the given database name first in a matching bucket name,
+// but then fall back to searching through configs in each bucket to try and find a config.
+func (sc *ServerContext) _fetchAndLoadDatabase(dbName string) (found bool, err error) {
 	found, dbConfig, err := sc.fetchDatabase(dbName)
 	if err != nil || !found {
 		return false, err
 	}
-	sc.applyConfigs(map[string]DatabaseConfig{dbName: *dbConfig})
+	sc._applyConfigs(map[string]DatabaseConfig{dbName: *dbConfig})
 
 	return true, nil
 }
 
 func (sc *ServerContext) fetchDatabase(dbName string) (found bool, dbConfig *DatabaseConfig, err error) {
+	select {
+	case <-sc.bootstrapContext.terminator:
```

Comment:
Do we need to worry about the bootstrapContext being terminated inside fetchDatabase after this initial check?

Reply:
I think this was added as a fix for when `handleCreateDB` was doing an async on-demand load and racing during test teardown. I don't think the check is required at all now that is synchronous.